packages/block-editor/src/components/inserter/reusable-blocks-tab.js

```diff
@@ -28,12 +30,15 @@ function ReusableBlocksSearchResults( {
 	);
 
 	const filteredItems = useMemo( () => {
+		if ( ! filterValue ) {
+			return items.filter( reusableBlockFilter );
+		}
 		return searchBlockItems(
 			items,
 			categories,
 			collections,
 			filterValue
-		).filter( ( { category } ) => category === 'reusable' );
+		).filter( reusableBlockFilter );
```

Comment:
Minor: if you filter the reusable block items before calling `searchBlockItems`, you'd be able to remove the duplication and inline `reusableBlockFilter`